packages/vitest/src/runtime/context.ts

```diff
@@ -44,22 +45,17 @@ export function withTimeout<T extends((...args: any[]) => any)>(
   }) as T
 }
 
-function makeTimeoutMsg(isHook: boolean, timeout: number) {
-  return `${isHook ? 'Hook' : 'Test'} timed out in ${timeout}ms.\nIf this is a long-running test, pass a timeout value as the last argument or configure it globally with "${isHook ? 'hookTimeout' : 'testTimeout'}".`
-}
+export function createTestContext(test: Test): TestContext {
+  const context = function () {
+    throw new Error('done() callback is deperated, use promise instead')
+  } as unknown as TestContext
 
-function ensureAsyncTest(fn: TestFunction): () => Awaitable<void> {
-  if (!fn.length)
-    return fn as () => Awaitable<void>
+  context.meta = test
+  context.expect = createExpect(test)
```

Comment:
Is the use case for this avoiding the need to import `expect`? Would it be possible to avoid creating a new `expect` instance for each test? Maybe perf isn't an issue, but IIUC the prev `expect` implementation would work as is if it is used while the test is running, no? (as `this.test` would point to the proper test)

Reply:
#551

Reply:
Changed to lazy evaluation

Reply:
Gotcha, lovely fix to avoid the overhead!